factory.js

```diff
@@ -1,25 +1,33 @@
 const states = require('./states.js');
 
-function factory(input) {
+function factory(dataType, link) {
```

Comment:
Hmm I think we should support not providing a link. We can just default it to null or something, and only add that object to the array in the constructors if link is not null?